spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -1030,9 +1036,18 @@ public Set<Long> getAllTaskExecutionIds(boolean onlyCompleted, String taskName)
 
 	@Override
 	public Integer getAllTaskExecutionsCount(boolean onlyCompleted, String taskName) {
-		return (int) (
-			onlyCompleted ? dataflowTaskExecutionQueryDao.getCompletedTaskExecutionCountByTaskName(taskName)
-				: dataflowTaskExecutionQueryDao.getTaskExecutionCountByTaskName(taskName)
-		);
+		return getAllTaskExecutionsCount(onlyCompleted, taskName, null);
+	}
+
+	@Override
+	public Integer getAllTaskExecutionsCount(boolean onlyCompleted, String taskName, Integer days) {
+		if (days != null) {
```

Comment:
@klopfdreh thanks for the updates to the cleanup path - looks great. I want to get this merged but I think the "what to do w/ END_TIME null when user specifies onlyCompleted='false'" still applies to the executions count path as well. 

I totally understand you have limited time - thank you for. your patience while we go through our back/forth :) .

Let's figure out what we want to do here and I am more than willing to make those changes and we can co-author this - unless you have the cycles to make another update. 

### Case 1
If `includeTasksEndedMinDaysAgo` is specified then that implies `onlyCompleted=true` - it has to be in order to compare the dates.
* the new API `getAllTaskExecutionsCount(boolean onlyCompleted, String taskName, Integer includeTasksEndedMinDaysAgo)` can have its `onlyCompleted` param removed and it would solve this case

### Case 2
If `includeTasksEndedMinDaysAgo` is NOT specified then `onlyCompleted=true|false` makes sense.
* the already existing API `getAllTaskExecutionsCount(boolean onlyCompleted, String taskName)` solves this one

Wdyt?

Reply:
You are completely right with the scenario if the END_TIME is null the onlyCompleted should still apply! Thanks your a lot for the time to review and for the hints!

Sadly I am busy by solving issues at our integration and I don’t know when I am able to revisit to make changes again.

I would appreciate it very much if you could assist me a bit with those changes.

As this feature very important - I guess also for a lot of users of SCDF - it would be very nice to see it integrated soon as your time allows it.

Thank you a lot for the help and the code reviews and of course for the assistance!

Reply:
> Thanks your a lot for the time to review and for the hints!

You are more than welcome - anytime.

> Sadly I am busy by solving issues at our integration and I don’t know when I am able to revisit to make changes again.
> I would appreciate it very much if you could assist me a bit with those changes.
> 

No worries @klopfdreh . I will get it updated and integrated this weekend. ☮️ and thanks again.